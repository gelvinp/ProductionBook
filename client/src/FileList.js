import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { List } from 'semantic-ui-react'

class FileList extends Component {
  render() {
    const { files, selectFile } = this.props
    const visibleFiles = files.map(file => (
      <List.Item onClick={() => selectFile(file.hash)} key={file.hash}>
        {file.name}
      </List.Item>
    ))
    return <List selection>{visibleFiles}</List>
  }
}

FileList.propTypes = {
  files: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      hash: PropTypes.string.isRequired
    })
  ).isRequired,
  selectFile: PropTypes.func.isRequired
}

export default FileList
