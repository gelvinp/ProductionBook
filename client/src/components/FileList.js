import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { List } from 'semantic-ui-react'

class FileList extends Component {
  render() {
    const { files, selectDocument } = this.props
    const visibleFiles = Object.entries(files).map(([uuid, file]) => (
      <List.Item key={`file-${uuid}`} onClick={() => selectDocument(uuid)}>
        <List.Icon name="file" />
        <List.Content>{file.name}</List.Content>
      </List.Item>
    ))
    return (
      <List.List className="ui selection relaxed">{visibleFiles}</List.List>
    )
  }
}

FileList.propTypes = {
  files: PropTypes.object.isRequired,
  selectDocument: PropTypes.func.isRequired,
}

export default FileList
