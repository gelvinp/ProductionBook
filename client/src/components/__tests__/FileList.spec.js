import React from 'react'
import FileList from '../FileList.js'
import { shallow } from 'enzyme'

describe('FileList', () => {
  it('Renders provided files', () => {
    const name = 'Sample File'
    const uuid = '1'
    const selectDocument = jest.fn()
    const files = {
      [uuid]: {
        name,
      },
    }
    const wrapper = shallow(
      <FileList files={files} selectDocument={selectDocument} />
    )
    expect(wrapper.find('ListItem').key()).toBe('file-1')
    expect(wrapper.find('ListIcon').props().name).toBe('file')
    expect(
      wrapper
        .find('ListContent')
        .childAt(0)
        .text()
    ).toBe(name)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('ListItem').simulate('click')
    expect(selectDocument.mock.calls.length).toBe(1)
  })
})
