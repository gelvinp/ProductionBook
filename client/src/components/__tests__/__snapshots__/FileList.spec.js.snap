// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList Renders provided files 1`] = `
<ListList
  className="ui selection relaxed"
>
  <ListItem
    key="file-1"
    onClick={[Function]}
  >
    <ListIcon
      name="file"
    />
    <ListContent>
      Sample File
    </ListContent>
  </ListItem>
</ListList>
`;
